src/Setup/Objective.php

```diff
@@ -54,4 +54,12 @@ public function getPreconditions(Environment $environment) : array;
      * @throw \RuntimeException if there are missing resources.
      */
     public function achieve(Environment $environment) : Environment;
+
+    /**
+     * Get to know whether the objective is applicable.
+     *
+     * Don't change the environment or act on services in the environment.
+     * In case of doubt whether the objective is applicable or not return true.
```

Comment:
```suggestion
     * Don't change the environment or cause changes on services in the environment.
     * Just check if this goal needs to be achieved, either currently or at all.
     * In case of doubt whether the objective is applicable or not return true.
```